### PR TITLE
PD-937: Adds "submitDisbaled" prop to ConfirmationModal

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-937](https://jira.mongodb.org/browse/PD-937)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx [hunk 1]
```diff
@@ -146,4 +146,16 @@ describe('packages/confirmation-modal', () => {
...
+
+  describe('submit is disabled when', () => {
+    test('"submitDisabled" prop is set', () => {
+      const { getByText } = renderModal({
+        open: true,
+        submitDisabled: true,
+      });
+
+      const confirmationButton = getByText('Confirm').closest('button');
+      expect(confirmationButton).toBeDisabled();
```
Comment (lazytype) [inline packages/confirmation-modal/src/ConfirmationModal.spec.tsx:158]: can you also add that clicking the button doesn't cause the modal to be closed?

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx [hunk 2]
```diff
@@ -146,4 +146,16 @@ describe('packages/confirmation-modal', () => {
       expect(confirmationButton).toBeDisabled();
     });
   });
+
+  describe('submit is disabled when', () => {
+    test('"submitDisabled" prop is set', () => {
```
Comment (lazytype) [inline packages/confirmation-modal/src/ConfirmationModal.spec.tsx:151]: would be also good to test this in combination with the `requiredInputText` prop

**Review by lazytype — CHANGES_REQUESTED** [review]

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/confirmation-modal/src/ConfirmationModal.spec.tsx:158]: 👍

File: packages/confirmation-modal/src/ConfirmationModal.spec.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/confirmation-modal/src/ConfirmationModal.spec.tsx:151]: 👍

**Review by lazytype — APPROVED** [review]

**Pull request merged**